lib/core/include/irods/process_stash.hpp

```diff
@@ -70,8 +73,10 @@ namespace irods::process_stash
     /// If \p _pred returns \p true, the entry is removed.
     /// \endparblock
     ///
+    /// \returns The number of elements removed.
+    ///
     /// \since 4.3.1
-    auto erase_if(const std::function<bool(const std::string&, const boost::any&)>& _pred) -> void;
+    auto erase_if(const std::function<bool(const std::string&, const boost::any&)>& _pred) -> std::unordered_map<std::string, boost::any>::size_type;
```

Comment:
Let's just return `std::size_t`. This exposes an implementation detail which doesn't need to be known by users of the lib.

Consider what happens if we switch the underlying container.